kedro/io/core.py

```diff
@@ -273,6 +273,15 @@ def _pretty_repr(self, object_description: dict[str, Any]) -> str:
         return f"{type(self).__module__}.{type(self).__name__}({', '.join(str_keys)})"
 
     def __repr__(self) -> str:
+        object_description = self._describe()
+        if not isinstance(object_description, dict) or not all(
+            isinstance(key, str) for key in object_description.keys()
```

Comment:
- In 278 we know that `object_description` is a `dict` because if it is not, the `or` expression would stop at 277 as return `True` from the left side of `or`.
- Therefore `.keys()` is redundant since default `for` over a `dict` is an iteration over its `keys` implicitly.

```suggestion
            isinstance(key, str) for key in object_description
```

Reply:
Agree with @yury-fedotov. NIT but I prefer the opposite logic since not A or not B is harder to reason about

I'd prefer this style since it follows the intend:
if A and B:
  do the normal describe()
else:
   something is wrong, show the warning.